ecs-cli/modules/config/config_v1.go

```diff
@@ -0,0 +1,142 @@
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package config
+
+import (
+	"fmt"
+	"os"
+
+	cli "github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+const (
+	configVersion               = "v1"
+	composeProjectNamePrefixKey = "compose-project-name-prefix"
+	composeServiceNamePrefixKey = "compose-service-name-prefix"
+	cfnStackNamePrefixKey       = "cfn-stack-name-prefix"
+)
+
+// CliConfig is the top level struct used to map to the yaml config.
+// Version: v1 = YAML formatted configs split profile and cluster
+type CliConfig struct {
+	Version                  string `yaml:"version"`
+	Cluster                  string `yaml:"cluster"`
+	AwsProfile               string `yaml:"aws_profile"`
+	Region                   string `yaml:"region"`
+	AwsAccessKey             string `yaml:"aws_access_key_id"`
+	AwsSecretKey             string `yaml:"aws_secret_access_key"`
+	ComposeProjectNamePrefix string `yaml:"compose-project-name-prefix"`
+	ComposeServiceNamePrefix string `yaml:"compose-service-name-prefix"`
+	CFNStackNamePrefix       string `yaml:"cfn-stack-name-prefix"`
+}
+
+// NewCliConfig creates a new instance of CliConfig from the cluster name.
+func NewCliConfig(cluster string) *CliConfig {
+	return &CliConfig{Cluster: cluster}
+}
+
+// ToAWSSession creates a new Session object from the CliConfig object.
+//
+// Region: Order of resolution
+//  1) Environment Variable - attempts to fetch the region from environment variables:
+//    a) AWS_REGION (OR)
+//    b) AWS_DEFAULT_REGION
+//  2) ECS Config - attempts to fetch the region from the ECS config file
+//  3) AWS Profile - attempts to use region from AWS profile name
+//    a) profile name from ECS config file (OR)
+//    b) AWS_PROFILE environment variable (OR)
+//    c) AWS_DEFAULT_PROFILE environment variable (defaults to 'default')
+//
+// Credentials: Order of resolution
+//  1) Environment Variable - attempts to fetch the credentials from environment variables:
+//   a) AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY (OR)
+//   b) AWS_ACCESS_KEY and AWS_SECRET_KEY
+//  2) ECS Config - attempts to fetch the credentials from the ECS config file
+//  3) AWS Profile - attempts to use credentials (aws_access_key_id, aws_secret_access_key) or assume_role (role_arn, source_profile) from AWS profile name
+//    a) profile name from ECS config file (OR)
+//    b) AWS_PROFILE environment variable (OR)
+//    c) AWS_DEFAULT_PROFILE environment variable (defaults to 'default')
+//  4) EC2 Instance role
+func (cfg *CliConfig) ToAWSSession() (*session.Session, error) {
+	svcConfig := aws.Config{}
```

Comment:
Good point- I'll refactor that. 

(Technically I did not write this code tho actually- As I stated in the about, it was suggested to me that I rearrange the code- so config.go became config_v1.go and includes everything that is used now- whereas config.go is the old stuff that is being kept around for backwards compatibility. Unfortunately this makes it look like I have changed a lot more code than I actually have. I think in retrospect I should not have changed the file names. The blame suggests this code was written 2 years ago by Uttara. My guess is that may be it was done this way to facilitate testing?)